datumaro/components/extractor.py

```diff
@@ -359,6 +360,69 @@ def as_polygon(self):
     def get_area(self):
         return 0
 
+
+@attrs
+class Cuboid3d(Annotation):
+    _type = AnnotationType.cuboid_3d
+    _points = attrib(type=list, default=None)
+    label = attrib(converter=attr.converters.optional(int),
+        default=None, kw_only=True)
+
+    @_points.validator
+    def _points_validator(self, attribute, points):
+        if points is None:
+            points = [0, 0, 0,  0, 0, 0,  1, 1, 1]
+        else:
+            assert len(points) == 3 + 3 + 3, points
+            points = [round(p, _COORDINATE_ROUNDING_DIGITS) for p in points]
+        self._points = points
+
+    # will be overridden by attrs, then will be overridden again by us
+    # attrs' method will be renamed to __attrs_init__
+    def __init__(self, position, rotation=None, scale=None, **kwargs):
+        assert len(position) == 3, position
+        if not rotation:
+            rotation = [0] * 3
+        if not scale:
+            scale = [1] * 3
+        kwargs.pop('points', None)
+        self.__attrs_init__(points=[*position, *rotation, *scale], **kwargs)
```

Comment:
`position` is a point, but `rotation` and `scale` surely aren't. Why do you store them in one list?

Reply:
For memory and processing efficiency. You're not completely true about rotation and scale, because they all are N-d vectors. It's like 4x4 quaternions.